src/common/snippets/src/kernel_executor_table.cpp

```diff
@@ -31,7 +31,7 @@ KernelExecutorTable::ExecTableState KernelExecutorTable::get_state() const {
     // Note: we need to clone configs when saving the state, since the configs still stored in the table can
     // be modified e.g. by calling update_by_expression();
     for (const auto& record : m_table)
-        result.emplace_back(std::make_pair(record.first, record.second->get_config()->clone()));
+        result.emplace_back(std::make_pair(record.first, record.second->get_config().get_clone_ptr()));
```

Comment:
I'm totally with you here, but unfortunately we need to work with pointers in the `state` because every executor has (or can have) it's own type of `Config`. So a pointer to the base class is the only option here.
Let me know if you have other ideas.

Reply:
Exactly!
I always forget that we work with template classes and shared pointer here...
Thank you